sys/include/registry/string_path.h

```diff
@@ -0,0 +1,183 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+    REGISTRY_STRING_PATH_TYPE_NAMESPACE,
+    REGISTRY_STRING_PATH_TYPE_SCHEMA,
+    REGISTRY_STRING_PATH_TYPE_INSTANCE,
+    REGISTRY_STRING_PATH_TYPE_GROUP,
+    REGISTRY_STRING_PATH_TYPE_PARAMETER,
+    REGISTRY_STRING_PATH_TYPE_GROUP_OR_PARAMETER,
+}registry_string_path_type_t;
+
+#include "registry.h"
+
+/**
+ * @brief Converts a registry namespace object to its string path representation.
+ *
+ * @param[in] namespace The registry namespace object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_namespace_string_path(const registry_namespace_t *namespace, char *path);
+
+/**
+ * @brief Converts a registry schema object to its string path representation.
+ *
+ * @param[in] schema The registry schema object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_schema_string_path(const registry_schema_t *schema, char *path);
+
+/**
+ * @brief Converts a registry instance object to its string path representation.
+ *
+ * @param[in] instance The registry instance object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_instance_string_path(const registry_instance_t *instance, char *path);
+
+/**
+ * @brief Converts a registry group object to its string path representation.
+ *
+ * @param[in] instance The registry instance object to which the group belongs.
+ * @param[in] group The registry group object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_group_string_path(const registry_instance_t *instance,
+                                  const registry_group_t *group, char *path);
+
+/**
+ * @brief Converts a registry parameter object to its string path representation.
+ *
+ * @param[in] instance The registry instance object to which the parameter belongs.
+ * @param[in] parameter The registry parameter object to convert.
+ * @param[out] path The buffer to store the resulting string path.
+ *
+ * @return Length of string path on success, non-zero on failure.
+ */
+int registry_to_parameter_string_path(const registry_instance_t *instance,
+                                      const registry_parameter_t *parameter, char *path);
+
+/**
+ * @brief Converts a string path to a registry namespace object.
+ *
+ * @param[in] path The string path to convert.
+ * @param[out] namespace A pointer to the registry namespace object to be created.
+ *
+ * @return 0 on success, non-zero on failure.
+ */
+int registry_from_namespace_string_path(const char *path, registry_namespace_t **namespace);
```

Comment:
I will think of this further once we agreed on how to handle the whole situation with all those registry_from_to_etc functions.

But I agree having only `_to_` seems attractive to me as well.

At that time I opted for the `_to_` and `_from_` because I could not think of a generic name for the basic registry pointer types.

Example:
```c
registry_to_string_path();
registry_from_string_path();

// vs.

registry_to_string_path();
registry_string_path_to_registry();

// vs.

registry_registry_to_string_path();
registry_string_path_to_registry();

// vs.

registry_pointer_to_string_path();
registry_string_path_to_pointer();
```

I am happy with none of them :cry:

Reply:
This is totally outdated and does not exist anymore.
So it is "fixed"